d3a_api_client/redis.py

```diff
@@ -0,0 +1,183 @@
+            Commands.DELETE_BID: f'{self._channel_prefix}/delete_bid',
+            Commands.LIST_OFFERS: f'{self._channel_prefix}/offers',
+            Commands.LIST_BIDS: f'{self._channel_prefix}/bids'
+        }
+
+    def wait_and_consume_command_response(self, command_type):
+        wait_until_timeout_blocking(lambda: command_type in self._blocking_command_responses)
+        command_output = self._blocking_command_responses.pop(command_type)
+        return command_output
+
+    def _generate_command_response_callback(self, command_type):
+        def command_received(msg):
+            message = json.loads(msg["data"])
+            if 'error' in message:
+                logging.error(f"Error when receiving {command_type} command response."
+                              f"Error output: {message}")
+                return
+            else:
+                self._blocking_command_responses[command_type] = message
+        return command_received
+
+    @registered_connection
+    def offer_energy(self, energy, price):
+        self.redis_db.publish(
+            self._command_topics[Commands.OFFER],
+            json.dumps({"energy": energy, "price": price})
+        )
+        return self.wait_and_consume_command_response(Commands.OFFER)
+
+    @registered_connection
+    def bid_energy(self, energy, price):
+        self.redis_db.publish(
+            self._command_topics[Commands.BID],
+            json.dumps({"energy": energy, "price": price})
+        )
+        return self.wait_and_consume_command_response(Commands.BID)
+
+    @registered_connection
+    def delete_offer(self, offer_id):
+        self.redis_db.publish(
+            self._command_topics[Commands.DELETE_OFFER],
+            json.dumps({"offer": offer_id})
+        )
+        return self.wait_and_consume_command_response(Commands.DELETE_OFFER)
+
+    @registered_connection
+    def delete_bid(self, bid_id):
+        self.redis_db.publish(
+            self._command_topics[Commands.DELETE_BID],
+            json.dumps({"bid": bid_id})
+        )
+        return self.wait_and_consume_command_response(Commands.DELETE_BID)
+
+    @registered_connection
+    def list_offers(self):
+        self.redis_db.publish(self._command_topics[Commands.LIST_OFFERS], json.dumps(""))
+        return self.wait_and_consume_command_response(Commands.LIST_OFFERS)
+
+    @registered_connection
+    def list_bids(self):
+        self.redis_db.publish(self._command_topics[Commands.LIST_BIDS], json.dumps(""))
+        return self.wait_and_consume_command_response(Commands.LIST_BIDS)
+
+    def _on_register(self, msg):
+        message = json.loads(msg["data"])
+        if 'available_publish_channels' not in message or \
+                'available_subscribe_channels' not in message:
+            raise RedisAPIException(f'Registration to the market {self.market_id} failed.')
+
+        logging.info(f"Client was registered to market: {message}")
```

Comment:
I suggest to filter the message for the market here or make the prefix more generic (AFAIU, message does not only contain the market name, right?)

Reply:
Ah, I see now that you have the self. market_id handy here.